### PR TITLE
Convert eval result to ADT

### PR DESCRIPTION
**Pull request by andyscott**
This initial commit will be broken (in the site project). I will fix it shortly.

### CODE REVIEW

File: core/runtime/src/main/scala/com/fortysevendeg/exercises/MethodEval.scala [hunk 1]
```diff
@@ -15,28 +15,43 @@ import cats.std.list._
 import cats.syntax.flatMap._
 import cats.syntax.traverse._
 
+object MethodEval {
+  sealed abstract class EvaluationResult extends Product with Serializable {
+    def didRun: Boolean
```
Comment (raulraja) [inline core/runtime/src/main/scala/com/fortysevendeg/exercises/MethodEval.scala]: What about a constructor arg?

**Comment by raulraja** [comment]
So far LGTM!

**Comment by andyscott** [comment]
- I'd like to get rid of the nested `Xor`s in the site projects. This requires that the ADT in the runtime project is made available in that scope or we create a parallel ADT for the site projects. I wasn't sure what route we wanted to go, so I left the Xors.
- This should fix the issue in #232. Now, the right projection of the eval result indicates a full success. The original logic for updating user progress should work correctly, as it originally assumed this about the Xor.

**Comment by raulraja** [comment]
@juanpedromoreno can you review and merge this?

**Comment by juanpedromoreno** [comment]
Looks good!

**Pull request merged**